### PR TITLE
Second Round of API and Cloud API refactorings

### PR DESCRIPTION
**Pull request by Salaboy**

### CODE REVIEW

File: activiti-cloud-services-runtime-bundle/activiti-cloud-services-core/src/main/java/org/activiti/cloud/services/core/commands/ClaimTaskCmdExecutor.java [hunk 1]
```diff
@@ -24,13 +25,13 @@ public ClaimTaskCmdExecutor(SecurityAwareTaskService securityAwareTaskService,
 
     @Override
     public String getHandledType() {
-        return TaskCommands.CLAIM_TASK.name();
+        return ClaimTaskPayload.class.getName();
```
[the remark this answers is not in the document]
Comment (Salaboy) [reply activiti-cloud-services-runtime-bundle/activiti-cloud-services-core/src/main/java/org/activiti/cloud/services/core/commands/ClaimTaskCmdExecutor.java]: @erdemedeiros yes please change

File: activiti-cloud-services-runtime-bundle/activiti-cloud-services-core/src/main/java/org/activiti/cloud/services/core/commands/CommandEndpoint.java [hunk 2]
```diff
@@ -31,7 +31,8 @@ public void consumeActivateProcessInstanceCmd(T cmd) {
     }
 
     private void processCommand(T cmd) {
```
[the remark this answers is not in the document]
Comment (Salaboy) [reply activiti-cloud-services-runtime-bundle/activiti-cloud-services-core/src/main/java/org/activiti/cloud/services/core/commands/CommandEndpoint.java]: @erdemedeiros yes please change

File: activiti-cloud-services-runtime-bundle/activiti-cloud-services-core/src/main/java/org/activiti/cloud/services/core/commands/CommandEndpoint.java [hunk 3]
```diff
@@ -31,7 +31,8 @@ public void consumeActivateProcessInstanceCmd(T cmd) {
...
-        CommandExecutor<T> cmdExecutor = commandExecutors.get(cmd.getCommandType().name());
+
+        CommandExecutor<T> cmdExecutor = commandExecutors.get(cmd.getClass().getName());
```
[the remark this answers is not in the document]
Comment (Salaboy) [reply activiti-cloud-services-runtime-bundle/activiti-cloud-services-core/src/main/java/org/activiti/cloud/services/core/commands/CommandEndpoint.java]: @erdemedeiros yes please change

File: activiti-cloud-services-runtime-bundle/activiti-cloud-services-core/src/main/java/org/activiti/cloud/services/core/commands/RemoveProcessVariablesCmdExecutor.java [hunk 4]
```diff
@@ -24,13 +24,14 @@ public RemoveProcessVariablesCmdExecutor(SecurityAwareProcessInstanceService sec
...
     }
 
     @Override
-    public void execute(RemoveProcessVariables cmd) {
-        securityAwareProcessInstanceService.removeProcessVariables(cmd);
-        RemoveProcessVariablesResultImpl cmdResult = new RemoveProcessVariablesResultImpl(cmd);
-        commandResults.send(MessageBuilder.withPayload(cmdResult).build());
+    public void execute(RemoveProcessVariablesPayload removeProcessVariablesPayload) {
+        securityAwareProcessInstanceService.removeProcessVariables(removeProcessVariablesPayload);
+        ProcessInstanceResult result = new ProcessInstanceResult(removeProcessVariablesPayload,
```
[the remark this answers is not in the document]
Comment (Salaboy) [reply activiti-cloud-services-runtime-bundle/activiti-cloud-services-core/src/main/java/org/activiti/cloud/services/core/commands/RemoveProcessVariablesCmdExecutor.java]: @erdemedeiros yes totally, please change

File: activiti-cloud-services-runtime-bundle/activiti-cloud-services-core/src/main/java/org/activiti/cloud/services/core/commands/SetProcessVariablesCmdExecutor.java [hunk 5]
```diff
@@ -24,14 +23,14 @@ public SetProcessVariablesCmdExecutor(SecurityAwareProcessInstanceService proces
...
     }
 
     @Override
-    public void execute(SetProcessVariables cmd) {
-        processInstanceService.setProcessVariables(cmd);
-        SetProcessVariablesResultImpl cmdResult = new SetProcessVariablesResultImpl(cmd);
-        commandResults.send(MessageBuilder.withPayload(cmdResult).build());
+    public void execute(SetProcessVariablesPayload setProcessVariablesPayload) {
+        processInstanceService.setProcessVariables(setProcessVariablesPayload);
+        ProcessInstanceResult result = new ProcessInstanceResult(setProcessVariablesPayload,
```
[the remark this answers is not in the document]
Comment (Salaboy) [reply activiti-cloud-services-runtime-bundle/activiti-cloud-services-core/src/main/java/org/activiti/cloud/services/core/commands/SetProcessVariablesCmdExecutor.java]: @erdemedeiros yes .. great idea

File: activiti-cloud-services-runtime-bundle/activiti-cloud-services-core/src/main/java/org/activiti/cloud/services/core/commands/SetTaskVariablesCmdExecutor.java [hunk 6]
```diff
@@ -24,13 +24,14 @@ public SetTaskVariablesCmdExecutor(SecurityAwareTaskService securityAwareTaskSer
...
     }
 
     @Override
-    public void execute(SetTaskVariables cmd) {
-        securityAwareTaskService.setTaskVariables(cmd);
-        SetTaskVariablesResultImpl cmdResult = new SetTaskVariablesResultImpl(cmd);
-        commandResults.send(MessageBuilder.withPayload(cmdResult).build());
+    public void execute(SetTaskVariablesPayload setTaskVariablesPayload) {
+        securityAwareTaskService.setTaskVariables(setTaskVariablesPayload);
+        TaskResult result = new TaskResult(setTaskVariablesPayload,
```
[the remark this answers is not in the document]
Comment (Salaboy) [reply activiti-cloud-services-runtime-bundle/activiti-cloud-services-core/src/main/java/org/activiti/cloud/services/core/commands/SetTaskVariablesCmdExecutor.java]: @erdemedeiros +1

File: activiti-cloud-services-runtime-bundle/activiti-cloud-services-core/src/main/java/org/activiti/cloud/services/core/commands/SignalCmdExecutor.java [hunk 7]
```diff
@@ -24,13 +23,14 @@ public SignalCmdExecutor(SecurityAwareProcessInstanceService processInstanceServ
...
     }
 
     @Override
-    public void execute(SendSignal cmd) {
-        processInstanceService.signal(cmd);
-        SendSignalResultImpl cmdResult = new SendSignalResultImpl(cmd);
-        commandResults.send(MessageBuilder.withPayload(cmdResult).build());
+    public void execute(SignalPayload signalPayload) {
+        processInstanceService.signal(signalPayload);
+        ProcessInstanceResult result = new ProcessInstanceResult(signalPayload,
```
[the remark this answers is not in the document]
Comment (Salaboy) [reply activiti-cloud-services-runtime-bundle/activiti-cloud-services-core/src/main/java/org/activiti/cloud/services/core/commands/SignalCmdExecutor.java]: @erdemedeiros +1

File: activiti-cloud-services-runtime-bundle/activiti-cloud-services-core/src/test/java/org/activiti/cloud/services/core/pageable/SecurityAwareProcessInstanceServiceTest.java [hunk 8]
```diff
@@ -179,36 +181,42 @@ public void shouldNotSuspendWithoutPermission() {
...
+                                                               UUID.randomUUID().toString(),
+                                                               "my-proc");
         given(processRuntime.processInstance(processInstance.getId())).willReturn(processInstance);
         given(securityService.canRead(processInstance.getProcessDefinitionKey())).willReturn(true);
-        FluentProcessDefinition def = buildProcessDefinition("my-proc");
-        when(processRuntime.processDefinitionById(processInstance.getProcessDefinitionId())).thenReturn(def);
-
-        when(securityService.canWrite(def.getKey())).thenReturn(hasWritePermission);
+        ProcessDefinition processDefinition = buildProcessDefinition("my-proc");
+//        when(processRuntime.processDefinitionById(processInstance.getProcessDefinitionId())).thenReturn(def);
```
[the remark this answers is not in the document]
Comment (Salaboy) [reply activiti-cloud-services-runtime-bundle/activiti-cloud-services-core/src/test/java/org/activiti/cloud/services/core/pageable/SecurityAwareProcessInstanceServiceTest.java]: @erdemedeiros please do

File: activiti-cloud-services-runtime-bundle/activiti-cloud-services-core/src/test/java/org/activiti/cloud/services/core/pageable/SecurityAwareTaskServiceTest.java [hunk 9]
```diff
@@ -65,93 +57,96 @@ public void setUp() {
...
-        FluentTask task = mock(FluentTask.class);
-        given(taskRuntime.task(cmd.getTaskId())).willReturn(task);
-
-        //when
-        taskService.setTaskVariablesLocal(cmd);
-
-        //when
-        verify(task).localVariables(cmd.getVariables());
-    }
+//    @Test
```
[the remark this answers is not in the document]
Comment (Salaboy) [reply activiti-cloud-services-runtime-bundle/activiti-cloud-services-core/src/test/java/org/activiti/cloud/services/core/pageable/SecurityAwareTaskServiceTest.java]: @erdemedeiros can you enable them and see if they are working or need too much work?

File: activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/cmdendpoint/CommandEndpointIT.java [hunk 10]
```diff
@@ -196,15 +196,15 @@ public void eventBasedStartProcessTests() throws Exception {
     private void completeTask(Task task) {
         Map<String, Object> variables = new HashMap<>();
 
-        CompleteTaskImpl completeTaskCmd = new CompleteTaskImpl(task.getId(),
-                                                                variables);
+        CompleteTaskPayload completeTaskCmd = TaskPayloadBuilder.complete().withTaskId(task.getId()).withVariables(
```
[the remark this answers is not in the document]
Comment (Salaboy) [reply activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/cmdendpoint/CommandEndpointIT.java]: @erdemedeiros yes.. as we discussed we can do that

File: activiti-cloud-services-runtime-bundle/activiti-cloud-services-core/src/main/java/org/activiti/cloud/services/core/commands/ClaimTaskCmdExecutor.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (erdemedeiros) [reply activiti-cloud-services-runtime-bundle/activiti-cloud-services-core/src/main/java/org/activiti/cloud/services/core/commands/ClaimTaskCmdExecutor.java]: Actually I've just realised the part can be kept as `class.getName` as it's only used internally to select the command executor: it's not used by jackson for serialization.

**Pull request merged**